### PR TITLE
VTN-24166 Flow Docs

### PR DESCRIPTION
**Pull request by aMacDawg**
Initial draft of "Building Flows" section nested in Developer section (there will likely be several different homes for this)

### CODE REVIEW

File: docs/developer/flow/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<!-- This is the Readme for the `Building Flows` sub-section of the Developer section on docs.veritone.com -->
+<!-- The general layout will read something like: Building Flows -> What is Flow -> Why Flow -> Creating Flows --> 
+<!-- How does it work? -> Quickstarts -> FAQ -->
+
+# Building Flows
+
+Now, with Veritone Flow, less technical users can create their own AI Solutions that incorporate Veritone's cognitive engines and real time services directly into daily business problems that were historically untenable with out an active intelligence participating.
```
Comment (SteveShaffer) [inline docs/developer/flow/README.md]: Let's remove any time-sensitive descriptors like "Now,". This should be able to stand the test of time.

File: docs/developer/flow/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## What is Flow?
+
+Veritone Flow has three components: 
+1. The Flow Designer where users can drag and drop shapes (or "nodes") that visualize discrete logic onto the Flow Designer canvas and test the logic using the Flow Designer runtime.
+1. The Veritone Real Time framework which serves as the production grade runtime for the Flow, running "as an engine," which in Veritone vocab, is just a fancy way of saying "custom logic that runs on Veritone's runtime"
```
Comment (SteveShaffer) [inline docs/developer/flow/README.md]: The Veritone Flow engine runtime where engines run in a horizontally scalable infrastructure in aiWARE's edge software.

File: docs/developer/flow/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+1. Managing the Flow application and related features as they are released to aiWARE users
```
Comment (SteveShaffer) [inline docs/developer/flow/README.md]: This is an awkward sentence and I'm not sure what it means.

File: docs/developer/flow/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+## Why do you need Flow?
+
+Flow enables you to rapidly develop AI solutions that let you tap into veritone's 400+ Cognitive engines across more than a dozen classes of AI cognition and incorporate that artificial intelligence into an existing business process that you want to automate.
```
Comment (SteveShaffer) [inline docs/developer/flow/README.md]: Can we use more direct language?  Like "Flow enables anyone to build business logic services without having to know how to code." or something like that?

File: docs/developer/flow/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+1. The Veritone Real Time framework which serves as the production grade runtime for the Flow, running "as an engine," which in Veritone vocab, is just a fancy way of saying "custom logic that runs on Veritone's runtime"
+1. Managing the Flow application and related features as they are released to aiWARE users
+
+## Why do you need Flow?
+
+Flow enables you to rapidly develop AI solutions that let you tap into veritone's 400+ Cognitive engines across more than a dozen classes of AI cognition and incorporate that artificial intelligence into an existing business process that you want to automate. 
+
+## What can you do with Flow?
+
+While Flow enables you to rapidly develop AI solutions that run on Veritone's real time and horizontally scaling framework, what you build is entirely up to you. Want to send out email alerts when a Face is recognized? You can use Flow to do that! Want to automate scraping the web for training data? You can do that too! You could even build a flow that texts your designated recipients and asks them to confirm if certain detections are accurate, and then setup logic to handle the recipient's text back!
```
Comment (SteveShaffer) [inline docs/developer/flow/README.md]: Please avoid use of the word "real-time." We just call it our edge computing infrastructure.  And if you want to make it sound fancy, you say "horizontally scalable."

File: docs/developer/flow/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+## How does it work?
+
+The Flow Designer is built with the Node-RED open source technology that is supported by the JavaScript foundation. At Veritone, we have taken this foundational open source technology and infused our own aiWARE palette of nodes that abstract away the underlying Veritone logic and data model, as well as make it easier for users to take advantage of the underlying power of aiWARE in a more approachable and less technically demanding format.
```
Comment (SteveShaffer) [inline docs/developer/flow/README.md]: Add links please

File: docs/developer/flow/README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+While Flow enables you to rapidly develop AI solutions that run on Veritone's real time and horizontally scaling framework, what you build is entirely up to you. Want to send out email alerts when a Face is recognized? You can use Flow to do that! Want to automate scraping the web for training data? You can do that too! You could even build a flow that texts your designated recipients and asks them to confirm if certain detections are accurate, and then setup logic to handle the recipient's text back!
+
+## How does it work?
+
+The Flow Designer is built with the Node-RED open source technology that is supported by the JavaScript foundation. At Veritone, we have taken this foundational open source technology and infused our own aiWARE palette of nodes that abstract away the underlying Veritone logic and data model, as well as make it easier for users to take advantage of the underlying power of aiWARE in a more approachable and less technically demanding format. 
+
+Make no mistake though, we certainly want to enable users who want to push the tool as far as they can!
+
+For more info on how Flow fits into the aiAWRE architecture, see the *Flow Engine* section of the docs.
```
Comment (SteveShaffer) [inline docs/developer/flow/README.md]: Add section then link to it

File: docs/developer/flow/README.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+## What can you do with Flow?
+
+While Flow enables you to rapidly develop AI solutions that run on Veritone's real time and horizontally scaling framework, what you build is entirely up to you. Want to send out email alerts when a Face is recognized? You can use Flow to do that! Want to automate scraping the web for training data? You can do that too! You could even build a flow that texts your designated recipients and asks them to confirm if certain detections are accurate, and then setup logic to handle the recipient's text back!
+
+## How does it work?
+
+The Flow Designer is built with the Node-RED open source technology that is supported by the JavaScript foundation. At Veritone, we have taken this foundational open source technology and infused our own aiWARE palette of nodes that abstract away the underlying Veritone logic and data model, as well as make it easier for users to take advantage of the underlying power of aiWARE in a more approachable and less technically demanding format. 
+
+Make no mistake though, we certainly want to enable users who want to push the tool as far as they can!
```
Comment (SteveShaffer) [inline docs/developer/flow/README.md]: I would remove this whole paragraph.  It's defensive and marketing fluff.

File: docs/developer/flow/faq.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
+# FAQ
+
+#### What should I do with Flow?
+
+Hrmmm, this is an interesting one! We recommend starting with this question: What kind of process do you or your customers still have to manually address with humans actively in the process loop? We think you will have some items that come to mind! You can use Flow to address these pain points and rapidly develop new solutions that automate more of these processes and move your end-users to the edge of the loop.
```
Comment (SteveShaffer) [inline docs/developer/flow/faq.md]: Can we use slightly less casual, conversational language?

File: docs/developer/flow/faq.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+#### How technical do I need to be to use Flow?
+
+Flow is a visual workflow tool that does use Node-RED's underlying technology, so some basic familiarity with coding principles would be very helpful but not a hard requirement. The Flow Designer is visuall driven, so coding is not required.
+
+If you are interested in going deeper, we recommend Codecademy's free course on Javascript or equivalent instruction to really drill into some of the Flow Designer's deeper features.
```
Comment (SteveShaffer) [inline docs/developer/flow/faq.md]: Add a link to the course

File: docs/developer/flow/faq.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+A fair amount, although we are looking where we can decrease the ramp needed. We have introduced several new aiWARE nodes specifically for abstracting away some knowledge of GraphQL to make it easier for users to jump straight into incorporating AI into their workflows.
+
+#### Can I expect to run my Flow on the Flow Designer Runtime for a long standing period of time?
+
+In short: no. The longer answer is that the Flow Designer is intended to serve as your testing environment. The Designer is not directly connected to Veritone's production run time, and so all of the testing is mock data. We have created (and will undoubtedly create more) several guides on how to create your flows so that you understand how to create your engine to run on Veritone's real time framework.
+
+#### What is a "Flow Engine?"
+
+This phrase serves as a description of the Flow once it has been deployed "as an Engine" to run on Veritone's realtime framework. Veritone engines take an input and produce an output that is managed by Veritone's realtime framework and run on Edge. You can check out the Veritone Engines docs for more details as well.
```
Comment (SteveShaffer) [inline docs/developer/flow/faq.md]: Remove "realtime" and replace with "edge" or something

File: docs/developer/flow/faq.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+#### What are the Chunk and Batch engine modes and which should I choose?
```
Comment (SteveShaffer) [inline docs/developer/flow/faq.md]: It's called "Segment" now, not "Chunk"

File: docs/developer/flow/faq.md [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+Chunk and Batch terms refer to "engine modes" for Veritone engines and refer to how Veritone's realtime framework
```
Comment (SteveShaffer) [inline docs/developer/flow/faq.md]: We should explain what each mode is and what it does.  And ideally link to a more global writeup on the subject too.

File: docs/developer/flow/faq.md (hunk 12 above)
Comment (aMacDawg) [reply docs/developer/flow/faq.md]: @SteveShaffer , is this still the case? I recall conversations around reverting to chunk but I acknowledge that Developer App currently displays the "Segment" option

File: docs/developer/flow/faq.md (hunk 9 above)
Comment (aMacDawg) [reply docs/developer/flow/faq.md]: Never thought I would see that comment ;-)

File: docs/developer/flow/faq.md (hunk 13 above)
Comment (aMacDawg) [reply docs/developer/flow/faq.md]: I'm happy to buff out the explanation in the context of Automate for 2019.30 release timeline and offer a more fleshed out generic version at a later date (within 2 weeks?)

**Pull request merged**